core/src/main/scala/org/apache/spark/util/Utils.scala

```diff
@@ -1651,7 +1651,7 @@ private[spark] object Utils extends Logging {
   /**
    * Format a Windows path such that it can be safely passed to a URI.
    */
-  def formatWindowsPath(path: String): String = path.replace("\\", "/")
+  def formatWindowsPath(path: String): String = path.replace("\\", "/").replace(" ", "%20")
```

Comment:
Ah, forgot to consider Linux since I have hardly ever seen the path with spaces in Linux.
I'll move this code to somewhere else.